alerting/src/main/kotlin/org/opensearch/alerting/workflow/CompositeWorkflowRunner.kt

```diff
@@ -80,87 +102,146 @@ object CompositeWorkflowRunner : WorkflowRunner() {
                     ?: throw AlertingException.wrap(
                         IllegalStateException("Chained finding monitor not found ${delegate.monitorId} for the workflow $workflow.id")
                     )
-
                 try {
                     indexToDocIds = monitorCtx.workflowService!!.getFindingDocIdsByExecutionId(chainedMonitor, executionId)
                 } catch (e: Exception) {
-                    val unwrappedException = ExceptionsHelper.unwrapCause(e) as Exception
-                    // If it is not IndexNotFound exception return the result
-                    if (unwrappedException.message?.contains("Configured indices are not found") == false) {
-                        logger.error("Failed to execute workflow. Error: ${e.message}", e)
-                        return workflowResult.copy(error = AlertingException.wrap(e))
-                    }
-                    // Log that finding index is not found and proceed with the execution
-                    logger.error("Finding index ${chainedMonitor.dataSources.findingsIndex} doesn't exist")
+                    logger.error("Failed to execute workflow due to failure in chained findings.  Error: ${e.message}", e)
+                    return WorkflowRunResult(
+                        workflow.id, workflow.name, emptyList(), workflowExecutionStartTime, Instant.now(), executionId,
+                        AlertingException.wrap(e)
+                    )
                 }
             }
-
             val workflowRunContext = WorkflowRunContext(
                 workflowId = workflowMetadata.workflowId,
                 workflowMetadataId = workflowMetadata.id,
                 chainedMonitorId = delegate.chainedMonitorFindings?.monitorId,
-                executionId = executionId,
-                matchingDocIdsPerIndex = indexToDocIds
+                matchingDocIdsPerIndex = indexToDocIds,
+                auditDelegateMonitorAlerts = if (workflow.auditDelegateMonitorAlerts == null) true
```

Comment:
You can do `workflow?.auditDelegateMonitorAlerts`

Reply:
it's optional

Reply:
Yea what I suggested was cleaner code to do the same thing

Reply:
keeping as is.. will take a follow up to clean this up